python/lsst/sims/coordUtils/Astrometry.py

```diff
@@ -888,18 +888,22 @@ def findChipName(self, xPupil=None, yPupil=None, ra=None, dec=None,
                 raise RuntimeError("No camera defined.  Cannot retrieve detector name.")
 
         chipNames = []
-        for x, y in zip(xPupil, yPupil):
-            cp = camera.makeCameraPoint(afwGeom.Point2D(x, y), PUPIL)
-            detList = [dd for dd in camera.findDetectors(cp) if dd.getType()==SCIENCE]
-            if len(detList) > 1 and not self.allow_multiple_chips:
-                raise RuntimeError("This method does not know how to deal with cameras where points can be"+
-                                   " on multiple detectors.  Override CameraCoords.get_chipName to add this.")
-            if not detList:
+        detList = camera.findDetectorsArray(xPupil, yPupil, PUPIL)
+        for det in detList:
+            if len(det)==0:
```

Comment:
This code would be clearer if you reserved "det" to mean "a Detector object". Thus perhaps:
detListList = camera.findDetectorsArray(...)
for detList in detListList:
    ...
   names = [det.getName() for det in detList...]